cider/exec/template/CaseIR.cpp

```diff
@@ -124,3 +124,111 @@ llvm::Value* CodeGenerator::codegenCase(const Analyzer::CaseExpr* case_expr,
   then_phi->addIncoming(else_lv, else_bb);
   return then_phi;
 }
+
+std::unique_ptr<CodegenColValues> CodeGenerator::codegenCaseExpr(
+    const Analyzer::CaseExpr* case_expr,
+    const CompilationOptions& co) {
+  AUTOMATIC_IR_METADATA(cgen_state_);
+  const auto case_ti = case_expr->get_type_info();
+  llvm::Type* case_llvm_type = nullptr;
+  bool is_real_str = false;
+  if (case_ti.is_integer() || case_ti.is_time() || case_ti.is_decimal()) {
+    case_llvm_type = get_int_type(get_bit_width(case_ti), cgen_state_->context_);
+  } else if (case_ti.is_fp()) {
+    case_llvm_type = case_ti.get_type() == kFLOAT
+                         ? llvm::Type::getFloatTy(cgen_state_->context_)
+                         : llvm::Type::getDoubleTy(cgen_state_->context_);
+  } else if (case_ti.is_string()) {
+    if (case_ti.get_compression() == kENCODING_DICT) {
+      case_llvm_type =
+          get_int_type(8 * case_ti.get_logical_size(), cgen_state_->context_);
+    } else {
+      is_real_str = true;
+      case_llvm_type = get_int_type(64, cgen_state_->context_);
+    }
+  } else if (case_ti.is_boolean()) {
+    case_llvm_type = get_int_type(8 * case_ti.get_logical_size(), cgen_state_->context_);
+  } else {
+    CIDER_THROW(CiderUnsupportedException,
+                fmt::format("case type is {}, not support", case_ti.get_type()));
+  }
+  CHECK(case_llvm_type);
+  const auto& else_ti = case_expr->get_else_expr()->get_type_info();
+  CHECK_EQ(else_ti.get_type(), case_ti.get_type());
+  std::unique_ptr<CodegenColValues> case_val =
+      codegenCaseExpr(case_expr, case_llvm_type, is_real_str, co);
+  if (is_real_str) {
+    // FIXME(haiwei): [POAE7-2457] ,blocking by [POAE7-2415]
+    CIDER_THROW(CiderCompileException,
```

Comment:
For String, the origin return is a vector contains multi values.  Will return a single ```std::unique_ptr<CodegenColValues>``` be not compatible with ```std::vector<llvm::Value*>```?

Reply:
This has been discussed with Jiang, will inherit CodegenColValues to contains std::vector<llvm::Value*> in [POAE7-2457] Support Case/IF Function for Arrow Format with String Function after [POAE7-2415] is supported.